### PR TITLE
Usage: add build description

### PR DESCRIPTION
**Pull request by dancesWithCycles**
Hi folks,
Thank you so much for providing and maintaining this repository. I came across this repo and stumbled over the build process. I did not find any description on how to build, deploy or test this repo. With some support I could add the missing description. Anyhow, my effort stopped so far after I called the following CLI instruction.

```shell
$ mvn clean package
```
Resulting in the following error.

```
[INFO] Scanning for projects...
Downloading from central: https://repo.maven.apache.org/maven2/org/rutebanken/superpom/superpom/1.0.28/superpom-1.0.28.pom
Downloading from conveyal: https://maven.conveyal.com/org/rutebanken/superpom/superpom/1.0.28/superpom-1.0.28.pom
Downloading from github: https://maven.pkg.github.com/entur/siri-protobuf-mapper/org/rutebanken/superpom/superpom/1.0.28/superpom-1.0.28.pom
Downloading from osgeo: https://repo.osgeo.org/repository/release/org/rutebanken/superpom/superpom/1.0.28/superpom-1.0.28.pom
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[FATAL] Non-resolvable parent POM for no.rutebanken.anshar:anshar:0.0.1-SNAPSHOT: Could not transfer artifact org.rutebanken.superpom:superpom:pom:1.0.28 from/to github (https://maven.pkg.github.com/entur/siri-protobuf-mapper): Authentication failed for https://maven.pkg.github.com/entur/siri-protobuf-mapper/org/rutebanken/superpom/superpom/1.0.28/superpom-1.0.28.pom 401 Unauthorized and 'parent.relativePath' points at wrong local POM @ line 21, column 11
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project no.rutebanken.anshar:anshar:0.0.1-SNAPSHOT (/home/begerad/git/github/anshar/pom.xml) has 1 error
[ERROR]     Non-resolvable parent POM for no.rutebanken.anshar:anshar:0.0.1-SNAPSHOT: Could not transfer artifact org.rutebanken.superpom:superpom:pom:1.0.28 from/to github (https://maven.pkg.github.com/entur/siri-protobuf-mapper): Authentication failed for https://maven.pkg.github.com/entur/siri-protobuf-mapper/org/rutebanken/superpom/superpom/1.0.28/superpom-1.0.28.pom 401 Unauthorized and 'parent.relativePath' points at wrong local POM @ line 21, column 11 -> [Help 2]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
[ERROR] [Help 2] http://cwiki.apache.org/confluence/display/MAVEN/UnresolvableModelException
```

I would appreciate any thoughts helping me building this repo.

Cheers!

### CODE REVIEW

**Comment by derhuerst** [comment]
Adding the steps required for a clean build to the `Dockerfile` would also be great! Or did you purposefully write it to depend on the non-Docker build?

**Comment by dancesWithCycles** [comment]
> Adding the steps required for a clean build to the `Dockerfile` would also be great! Or did you purposefully write it to depend on the non-Docker build?

Hi @derhuerst ,
How are you doing? Indeed, I do have a slight favour for a non-Docker build. Anyhow, in the first step I would appreciate any build procedure to get this repo building and running, even Docker.

The [Dockerfile](https://github.com/entur/anshar/blob/master/Dockerfile) looks quite short. Without having much knowledge about Docker, I wonder it that is enough for a Docker build.

Cheers!

**Comment by derhuerst** [comment]
> I do have a slight favour for a non-Docker build.

The two ways of building a deployable artifact (building a `.jar`, and building a Docker image running a `.jar`) are complementary, rather.

> The [Dockerfile](https://github.com/entur/anshar/blob/master/Dockerfile) looks quite short. Without having much knowledge about Docker, I wonder it that is enough for a Docker build.

Well it is enough to get a working Docker image, but usually you would run the build steps in the Dockerfile. With `.jar`s being platform-independent, that doesn't matter much though.

TLDR: Yes, we need a proper build script/command for this project.

**Comment by dancesWithCycles** [comment]
> TLDR: Yes, we need a proper build script/command for this project.

Hi @derhuerst ,
Do you personally know anyone that is active in this repo to invite into our discussion?

**Comment by lassetyr** [comment]
Hi, and thanks for your interest.

The build appears to fail because of some dependencies to other components that are currently only released to an internal artifactory. These are also public on github, so they could be made available on Maven Central. A "quick-fix" would be to check them out and build them locally, but I'll look into the possibility to release them properly. I agree that `mvn clean package` should suffice to build this.

**Comment by dancesWithCycles** [comment]
Hi folks,
Next to a build description, also a deployment, integration and test description would help this repository. Calling on the current main branch

```
$ mvn clean package -Dmaven.test.skip=true
$ java -jar target/anshar-0.0.1-SNAPSHOT.jar
```

brings the process to a straight halt with this response.

```$ java -jar target/anshar-0.0.1-SNAPSHOT.jar
  ___            _                
 / _ \          | |               
/ /_\ \_ __  ___| |__   __ _ _ __ 
|  _  | '_ \/ __| '_ \ / _` | '__|
| | | | | | \__ \ | | | (_| | |   
\_| |_/_| |_|___/_| |_|\__,_|_|

{"serviceContext":{"service":"anshar"},"message":"Starting App v0.0.1-SNAPSHOT using Java 11.0.16 on ltpt490 with PID 10049 (/home/begerad/git/sib/anshar/target/anshar-0.0.1-SNAPSHOT.jar started by begerad in /home/begerad/git/sib/anshar)\n","severity":"INFO","breadcrumb":"","correlationId":"","threadId":"main","reportLocation":{"filePath":"no.rutebanken.anshar.App","lineNumber":"55","functionName":"logStarting"}}
{"serviceContext":{"service":"anshar"},"message":"The following 1 profile is active: \"in-memory-blobstore\"\n","severity":"INFO","breadcrumb":"","correlationId":"","threadId":"main","reportLocation":{"filePath":"no.rutebanken.anshar.App","lineNumber":"640","functionName":"logStartupProfileInfo"}}
{"serviceContext":{"service":"anshar"},"message":"Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanDefinitionStoreException: Failed to parse configuration class [no.rutebanken.anshar.subscription.SubscriptionConfig]; nested exception is java.lang.IllegalStateException: java.io.FileNotFoundException: /etc/anshar-config/subscriptions.yml\n","severity":"WARN","breadcrumb":"","correlationId":"","threadId":"main","reportLocation":{"filePath":"org.springframework.context.annotation.AnnotationConfigApplicationContext","lineNumber":"591","functionName":"refresh"}}
{"serviceContext":{"service":"anshar"},"message":"Application run failed\norg.springframework.beans.factory.BeanDefinitionStoreException: Failed to parse configuration class [no.rutebanken.anshar.subscription.SubscriptionConfig]; nested exception is java.lang.IllegalStateException: java.io.FileNotFoundException: /etc/anshar-config/subscriptions.yml\n\tat org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:189)\n\tat org.springframework.context.annotation.ConfigurationClassPostProcessor.processConfigBeanDefinitions(ConfigurationClassPostProcessor.java:331)\n\tat org.springframework.context.annotation.ConfigurationClassPostProcessor.postProcessBeanDefinitionRegistry(ConfigurationClassPostProcessor.java:247)\n\tat org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanDefinitionRegistryPostProcessors(PostProcessorRegistrationDelegate.java:311)\n\tat org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:112)\n\tat org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:746)\n\tat org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:564)\n\tat org.springframework.boot.SpringApplication.refresh(SpringApplication.java:734)\n\tat org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:408)\n\tat org.springframework.boot.SpringApplication.run(SpringApplication.java:308)\n\tat org.springframework.boot.SpringApplication.run(SpringApplication.java:1306)\n\tat org.springframework.boot.SpringApplication.run(SpringApplication.java:1295)\n\tat no.rutebanken.anshar.App.main(App.java:32)\n\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)\n\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n\tat java.base/java.lang.reflect.Method.invoke(Method.java:566)\n\tat org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)\n\tat org.springframework.boot.loader.Launcher.launch(Launcher.java:108)\n\tat org.springframework.boot.loader.Launcher.launch(Launcher.java:58)\n\tat org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:88)\nCaused by: java.lang.IllegalStateException: java.io.FileNotFoundException: /etc/anshar-config/subscriptions.yml\n\tat org.springframework.beans.factory.config.YamlProcessor.handleProcessError(YamlProcessor.java:221)\n\tat org.springframework.beans.factory.config.YamlProcessor.process(YamlProcessor.java:213)\n\tat org.springframework.beans.factory.config.YamlProcessor.process(YamlProcessor.java:166)\n\tat org.springframework.boot.env.OriginTrackedYamlLoader.load(OriginTrackedYamlLoader.java:84)\n\tat org.springframework.boot.env.YamlPropertySourceLoader.load(YamlPropertySourceLoader.java:50)\n\tat no.rutebanken.anshar.subscription.YamlPropertySourceFactory.createPropertySource(YamlPropertySourceFactory.java:51)\n\tat org.springframework.context.annotation.ConfigurationClassParser.processPropertySource(ConfigurationClassParser.java:463)\n\tat org.springframework.context.annotation.ConfigurationClassParser.doProcessConfigurationClass(ConfigurationClassParser.java:280)\n\tat org.springframework.context.annotation.ConfigurationClassParser.processConfigurationClass(ConfigurationClassParser.java:250)\n\tat org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:207)\n\tat org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:175)\n\t... 20 common frames omitted\nCaused by: java.io.FileNotFoundException: /etc/anshar-config/subscriptions.yml\n\tat org.springframework.core.io.FileSystemResource.getInputStream(FileSystemResource.java:189)\n\tat org.springframework.beans.factory.config.YamlProcessor.process(YamlProcessor.java:197)\n\t... 29 common frames omitted\n","severity":"ERROR","breadcrumb":"","correlationId":"","threadId":"main","reportLocation":{"filePath":"org.springframework.boot.SpringApplication","lineNumber":"824","functionName":"reportFailure"}}
````

Do you mind providing insides on how to integrate, run and test this repository?

Cheers!

**Comment by lassetyr** [comment]
Hi,
I have updated the default config and added some simple instructions to build/run this as-is (Commit: https://github.com/entur/anshar/commit/290cd35fb11a18ab09c1e81c5546a95849701186) 

Could you please try again now?

**Comment by dancesWithCycles** [comment]
Hi Lasse @lassetyr ,
Thx for the changes. After pulling them and running

```
mvn clean package -Dmaven.test.skip=true
java -jar target/anshar-0.0.1-SNAPSHOT.jar
```

Anshar was launched and running. Eventually I observed that the port ``8012`` was not served by the Java process. With ``` >> log.txt 2>&1``` I created the following log file.

[log.txt](https://github.com/entur/anshar/files/9470146/log.txt)

That log file contains several error messages I can not classify at the first glimpse. Does those errors make any sense to you?

Cheers!
